src/Notifier.cpp

```diff
@@ -52,8 +54,8 @@ bool Notifier::Initialize(const size_t handshakeCount) {
       gQueue.reset(new Shotgun);
       try {
          gQueue->Aim(GetNotifierQueueName());
-      } catch (std::exception& e) {
-         LOG(WARNING) << "Caught exception: " << e.what();
+      } catch (std::string& e) {
```

Comment:
why would you use a string as exception instead of the standard std::exception?

In C++ you can throw anything. However, I  have yet to see a good reason for throwing anything but the std::exception class (or a subclass of it).

Please explain the reasoning and use case for this.